systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaUtils.java

```diff
@@ -354,4 +357,21 @@ public static String getVersionFromKafkaPodLibs(String kafkaPodName) {
             command
         ).out().trim();
     }
+
+    public static void waitForKafkaDeletion(String kafkaClusterName) {
+        LOGGER.info("Waiting for deletion of Kafka:{}", kafkaClusterName);
+        TestUtils.waitFor("Kafka deletion " + kafkaClusterName, Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS, DELETION_TIMEOUT,
+            () -> {
+                if (KafkaResource.kafkaClient().inNamespace(kubeClient().getNamespace()).withName(kafkaClusterName).get() == null &&
+                    kubeClient().getStatefulSet(KafkaResources.kafkaStatefulSetName(kafkaClusterName)) == null &&
+                    kubeClient().getStatefulSet(KafkaResources.zookeeperStatefulSetName(kafkaClusterName)) == null &&
+                    kubeClient().getDeployment(KafkaResources.entityOperatorDeploymentName(kafkaClusterName)) == null) {
+                    return true;
+                } else {
+                    KafkaResource.kafkaClient().inNamespace(kubeClient().getNamespace()).withName(kafkaClusterName).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete();
```

Comment:
I wonder if we should use here cmd client instead of fabric8 to avoid some issues with future upgrades. It's more just a question/idea. WDYT?

Reply:
This was what I thought too. Problem here was that when I did `cmdKubeClient().deleteByName(Kafka.RESOURCE_KIND, kafkaClusterName)`, only the Kafka pods and statefulsets were deleted.. unfortunately there are no `.withPropagationPolicy(DeletionPropagation.FOREGROUND)` or deprecated `.cascade(true)`... So I chose this way.. I will take a look on some better solution.

Reply:
The `kubectl` tools has the `--cascade` and `--wait` options which should cover the options from the Fabric8 client..

Reply:
Yep, that will be it. Let me try it and change it :) thanks @Frawless and @scholzj

Reply:
So how I thought, again problem on my side :smile: the `cmdKubeClient().deleteByName(Kafka.RESOURCE_KIND, kafkaClusterName)` is working! I'm gonna change it :) thanks again